src/main/java/net/atos/entng/calendar/controllers/CalendarController.java

```diff
@@ -151,15 +160,78 @@ public void deleteCalendar(HttpServerRequest request) {
+                    createFuture(user, body)
+                            .compose(calendarId -> {
+                                String host = getHost(request);
+                                String i18nLang = I18n.acceptLanguage(request);
+                                return calendarHelper.externalCalendarFirstSync(calendarId.getString(Field._ID, null),
+                                        user, host, i18nLang, ExternalICalEventBusActions.POST.method());
+                            })
+                            .onSuccess(result -> {
+                                Renders.ok(request);
+                            })
+                            .onFailure(error -> {
+                                String message = String.format("[Calendar@%s::importExternalCalendar] An error has occured" +
+                                        " during calendar sync: %s", this.getClass().getSimpleName(), error.getMessage());
+                                log.error(message, error.getMessage());
+                                renderError(request);
+                            });
+                });
+            } else {
+                unauthorized(request);
+            }
+        });
+    }
+
+    Future<JsonObject> createFuture(UserInfos user, JsonObject body) {
+        Promise<JsonObject> promise = Promise.promise();
+
+        crudService.create(body, user, r -> {
+            if (r.isLeft()) {
+                String message = String.format("[Calendar@%s::createFuture] An error has occured" +
+                        " during calendar creation: %s", this.getClass().getSimpleName(), r.left().getValue());
+                log.error(message, r.left().getValue());
+                promise.fail(message);
+            } else {
+                promise.complete(r.right().getValue());
+            }
+        });
+
+        return promise.future();
+    }
+
+    @Put("/:id/url")
+    @SecuredAction(Rights.UPDATE)
     @Trace(Actions.SYNC_EXTERNAL_CALENDAR)
     public void syncExternalCalendar(final HttpServerRequest request) {
         String calendar = request.params().get(Field.ID);
         String url = request.params().get(Field.URL);
         //processing method here
     }
 
+    @Get("/:id/url")
+    @SecuredAction(Rights.CHECKUPDATE)
+    @Trace(Actions.CHECK_EXTERNAL_CALENDAR)
+    public void checkSyncExternalCalendar(final HttpServerRequest request) {
+        String calendarId = request.params().get(Field.ID);
+        calendarService.checkBooleanField(calendarId, Field.ISUPDATING)
+                .onSuccess(result -> {
+                    Renders.renderJson(request, new JsonObject().put(Field.ISUPDATING, result));
+                })
+                .onFailure(err -> {
+                    log.error("[Calendar@CalendarController::checkSyncExternalCalendar]: an error has occurred while checking calendar: ",
+                            err.getMessage());
+                    Renders.badRequest(request);
```

Comment:
Ca serait plus un renderError puisque tu vas tomber dans une exception (ton onFailure)
```suggestion
                   Renders.renderError(request);
```